includes/class-amp-theme-support.php

```diff
@@ -829,7 +828,21 @@ protected static function ensure_required_markup( DOMDocument $dom ) {
 			) );
 			$head->insertBefore( $meta_viewport, $meta_charset->nextSibling );
 		}
-
+		// Prevent schema.org duplicates.
+		$schema_org_script = null;
+		foreach ( $head->getElementsByTagName( 'script' ) as $script ) {
+			if ( 'application/ld+json' === $script->getAttribute( 'type' ) && preg_match( '/{"@context":"https?:[\\\]\/[\\\]\/schema\.org/', $script->nodeValue ) ) {
```

Comment:
The regular expression here could be brittle. There's no guarantee that the `@context` will be the first key, and there's no guarantee there wouldn't be whitespace between the strings. I think doing a simple `false !== strpos( $script->nodeValue, 'schema.org' )` could be sufficient here.

Reply:
We may not even need to look at the value. Just checking if the type is `application/ld+json` may be just fine.

Reply:
Hi @westonruter 

I opted to search for both the attribute and the 'schema.org' substring as you mentioned it should check for any JSON-LD script in the head which mentions 'schema.org' in any form.